google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionImpl.java

```diff
@@ -298,6 +318,14 @@ public CommitResponse writeAtLeastOnceWithOptions(
     }
   }
 
+  private XGoogSpannerRequestId reqIdOrFresh(Options options) {
+    XGoogSpannerRequestId reqId = options.reqId();
+    if (reqId == null) {
+      reqId = this.getRequestIdCreator().nextRequestId(1 /* TODO: channelId */, 0);
```

Comment:
First attempt number should be 1

Reply:
So the idea is that I manually increase the attempt number before every invocation hence why it is 0. In the next PR though, I'll refine it and ensure that the attempts are only increased for the respective errors UNAVAILABLE and INTERNAL then nthRequest increased for ABORTED and any other retries.